File: 0217-contains-duplicate/0217-contains-duplicate.cpp
class Solution
{
public:
    bool containsDuplicate(vector<int> &nums)
    {
        sort(nums.begin(), nums.end());
        for (int i = 1; i < nums.size(); i++)
        {
            if (nums[i-1] == nums[i])
            {
                return true;
            }
        }
        return false;
    }
};

File: Python/0217-contains-duplicate.py
# time complexity: O(nlogn)
# space complexity: O(n)
from typing import List


class Solution:
    def containsDuplicate(self, nums: List[int]) -> bool:
        nums.sort()
        for i in range(len(nums)-1):
            if nums[i] == nums[i+1]:
                return True
        return False


Input = [1, 2, 3]

print(Solution().containsDuplicate(Input))
